pkg/controllers/managementuser/rbac/roletemplates/prtb_handler.go

```diff
@@ -0,0 +1,224 @@
+package roletemplates
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	mgmtv3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/rbac"
+	"github.com/rancher/rancher/pkg/types/config"
+	corev1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+	rbacv1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/rbac/v1"
+	"github.com/rancher/wrangler/v3/pkg/name"
+	v1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	prtbOwnerLabel      = "authz.cluster.cattle.io/prtb-owner"
+	projectIDAnnotation = "field.cattle.io/projectId"
+)
+
+type prtbHandler struct {
+	impersonationHandler impersonationHandler
+	crClient             rbacv1.ClusterRoleController
+	rtClient             mgmtv3.RoleTemplateController
+	nsClient             corev1.NamespaceController
+	rbClient             rbacv1.RoleBindingClient
+}
+
+func newPRTBHandler(uc *config.UserContext) *prtbHandler {
+	return &prtbHandler{
+		impersonationHandler: impersonationHandler{
+			userContext: uc,
+			crClient:    uc.RBACw.ClusterRole(),
+			crtbClient:  uc.Management.Wrangler.Mgmt.ClusterRoleTemplateBinding(),
+			prtbClient:  uc.Management.Wrangler.Mgmt.ProjectRoleTemplateBinding(),
+		},
```

Comment:
Would it make sense to add a constructor for the `impersonationHandler` and avoid picking every client here directly?